### ISSUE TITLE
`UnnamedParameterUse` should not be reported for java functions

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#7437]
## Expected Behavior
`notifyItemRangeChanged(0, currentList.size)` should not report any warning

## Observed Behavior
`notifyItemRangeChanged(0, currentList.size)` is reporting warning
Though using the intention from idea `Add names to call arguments` is converting the fun to `notifyItemRangeChanged(/* positionStart = */ 0, /* itemCount = */ currentList.size)`

## Steps to Reproduce
Happening in latest dev

**Comment by atulgpt** [comment]
There is already a TC for checking non Kotlin fun at https://github.com/detekt/detekt/blob/d4db424082b6b1b35ac6fb28990acfbf6616e602/detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnnamedParameterUseSpec.kt#L208

not sure what is getting missed

**Comment by 3flex** [comment]
Can you try with debug enabled to see if there are compiler errors?

Can you share a build scan?

Which Gradle task are you executing?

**Comment by atulgpt** [comment]
> Can you try with debug enabled to see if there are compiler errors?

Ran `./gradlew :app:detektStageDebug --debug > logs.txt` and attaching the logs.txt here(I don't think any error happened as suppressing is working fine)
[logs.txt](https://github.com/user-attachments/files/16099473/logs.txt)

> Can you share a build scan?
https://scans.gradle.com/s/2p7w6aj6uvtca

> Which Gradle task are you executing?
Detekt with TR `/gradlew :app:detektStageDebug`

**Comment by 3flex** [comment]
Thanks for the build scan.

This indicates there are issues with the task setup: https://scans.gradle.com/s/2p7w6aj6uvtca/console-log/task/:app:detektStageDebug?page=1#L14-L85

Are you using latest snapshot of the detekt Gradle plugin? If yes, and the task was registered by detekt Gradle plugin and not configured manually by you, then it should autoconfigure correctly. I would need to see the build scripts to troubleshoot further.

**Comment by atulgpt** [comment]
My `libs.version.toml` is as follow

```toml
gradlePlugins-detekt = "main-SNAPSHOT"

[libraries]
detekt-ktlint = { module = "io.gitlab.arturbosch.detekt:detekt-formatting", version.ref = "gradlePlugins-detekt" }

[plugins]
# linter or formatter
detekt = { id = "io.gitlab.arturbosch.detekt", version.ref = "gradlePlugins-detekt" }
```

My app level `build.gradle.kts` is as follow

```build.gradle.kts
plugins {
    id("android-application")
    kotlin("kapt")
    id("org.jetbrains.kotlin.plugin.parcelize")
    id("com.google.gms.google-services")
    id("androidx.navigation.safeargs.kotlin")
    id("kotlinx-serialization")
    alias(libs.plugins.detekt)
}

detekt {
    // If set to `true` the build does not fail when the
    // maxIssues count was reached. Defaults to `false`.
    ignoreFailures = false
    // point to your custom config defining rules to run, overwriting default behavior
    config.setFrom(files("$rootDir/AceLibs/config/detekt/detekt.yml"))
    baseline =
        file("$projectDir/config/baseline.xml") // a way of suppressing issues before introducing detekt
    // Specify the base path for file paths in the formatted reports.
    // If not set, all file paths reported will be absolute file path.
    basePath.set(projectDir)
    // Adds debug output during task execution. `false` by default.
    debug = true
    parallel = true
    buildUponDefaultConfig = true
    autoCorrect = true
    enableCompilerPlugin = true
}

tasks {
    withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
        jvmTarget = "1.8"
        reports {
            // observe findings in your browser with structure and code snippets
            html.required.set(true)
            // similar to the console output, contains issue signature to manually edit baseline files
            txt.required.set(false)
            sarif.required.set(false)
            xml.required.set(false)
        }
        exclude { it.file.path.contains("build") }
    }
}
```

**Comment by 3flex** [comment]
I might need a complete reproducer to look into it further... nothing sticks out at the moment. You could try not excluding the `build` folder too - perhaps the source that's required for the rule to work properly is missing thanks to that exclusion?

**Comment by BraisGabin** [comment]
@3flex do we have an issue to track that limitation? Maybe we should just remove the exclude from the client side and allow to define exclusions at the root of our config.yaml. This way we use all the information that we can get from the code but we don't raise any issue on those files.

**Comment by 3flex** [comment]
It shouldn't affect analysis because all required compiled classes should be on the analysis classpath already even if some files are excluded from analysis. There's not enough information to know what exactly is going on.

**Comment by atulgpt** [comment]
> You could try not excluding the build folder too - perhaps the source that's required for the rule to work properly is missing thanks to that exclusion?

Hi @3flex I tried removing the `exclude { it.file.path.contains("build") }` but the `notifyItemRangeChanged(0, currentList.size)` still present(since build is now not excluded all the errors in generated files in build also starts getting reported)

I will try to create the reproducer
